storage/diskmetricstore.go

```diff
@@ -211,6 +211,32 @@ func (dms *DiskMetricStore) processWriteRequest(wr WriteRequest) {
 			}
 			dms.metricGroups[key] = group
 		}
+
+		if wr.Lifetime != 0 {
+			// this is needed for closure awareness
+			n := name
+
+			// schedule timeout
+			time.AfterFunc( wr.Lifetime, func () {
```

Comment:
I think at medium scale this will run into big issues. Essentially we spawn a new goroutine on each request. While they are generally cheap to have around if they are inactive, this could cause problems.

A periodic garbage gollection plus filterting on read would seem like a safer approach.